src/Entity/Book.php

```diff
@@ -50,15 +48,90 @@ class Book
     #[ORM\OneToMany(mappedBy: 'book', targetEntity: Review::class)]
     public Collection $reviews;
 
+    #[ORM\ManyToMany(targetEntity: Category::class, inversedBy: 'books')]
+    public Collection $categories;
+
     public function __construct()
     {
         $this->physicalBooks = new ArrayCollection();
         $this->reviews = new ArrayCollection();
         $this->borrows = new ArrayCollection();
+        $this->id = Uuid::v4();
+        $this->categories = new ArrayCollection();
     }
 
     public function getId(): Uuid
     {
         return $this->id;
     }
+ 
+    public function getWriterFirstname(): string
+    {
+        return $this->writerFirstname;
+    }
+
+    public function setWriterFirstname(string $writerFirstname)
```

Comment:
```suggestion
    public function setWriterFirstname(string $writerFirstname): void
```


Same for all setters